Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1355,10 +1358,13 @@ RefPtr<API::UserInitiatedAction> WebProcessProxy::userInitiatedActivity(std::opt
     return userInitiatedActivity(identifier);
 }
 
-void WebProcessProxy::consumeIfNotVerifiablyFromUIProcess(API::UserInitiatedAction& action, std::optional<WTF::UUID> authToken)
+void WebProcessProxy::consumeIfNotVerifiablyFromUIProcess(PageIdentifier pageID, API::UserInitiatedAction& action, std::optional<WTF::UUID> authToken)
 {
-    if (authToken && m_userInitiatedActionByAuthorizationTokenMap.remove(*authToken))
+    auto authorizationTokenMapByPage = m_userInitiatedActionByAuthorizationTokenMap.get(pageID);
+    if (authToken && authorizationTokenMapByPage.contains(*authToken)) {
+        m_userInitiatedActionByAuthorizationTokenMap.remove(pageID);
```

Comment:
If you used find() above, you'd have an iterator to pass to remove() here which would be more efficient too.